opa-manager-operator/tests/integration/test_charm.py

```diff
@@ -0,0 +1,115 @@
+import logging
+from pathlib import Path
+import os
+import time
+import pytest
+import yaml
+import re
+from kubernetes import client, config
+from kubernetes.client.rest import ApiException
+from jinja2 import Template
+
+log = logging.getLogger(__name__)
+
+meta = yaml.safe_load(Path("metadata.yaml").read_text())
+
+KUBECONFIG = os.environ.get("TESTING_KUBECONFIG", "~/.kube/config")
+
+config.load_kube_config(KUBECONFIG)  # TODO: check how to use this with designated file
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    charm = await ops_test.build_charm(".")
+    # Due to: https://github.com/juju/python-libjuju/issues/515
+    #  We have to use the k8s API to wait, we cannot use:
+    # ops_test.model.applications['gatekeeper'].units[0].workload_status
+    model_name = ops_test._default_model_name
```

Comment:
The model name failed for me since I'm running the test with:
```
tox -e integration -- --cloud k8s --model k8s --keep-models
```

I ended up with a randomly generated model name, instead of the `k8s` model that the test actually used. I suggest getting the model name this way instead:

```suggestion
    model_name = ops_test.model_name
```